state-chain/chains/src/sol/sol_tx_core.rs

```diff
@@ -773,6 +773,61 @@ impl From<Hash> for SolHash {
 	}
 }
 
+#[derive(Encode, Decode, TypeInfo, Serialize, Deserialize, Debug, Clone)]
+pub struct CcmAddress {
+	pubkey: Pubkey,
+	is_writable: bool,
+}
+
+impl From<CcmAddress> for AccountMeta {
+	fn from(from: CcmAddress) -> Self {
+		match from.is_writable {
+			true => AccountMeta::new(from.pubkey, false),
+			false => AccountMeta::new_readonly(from.pubkey, false),
+		}
+	}
+}
+
+#[derive(Encode, Decode, TypeInfo, Serialize, Deserialize, Debug, Clone)]
+pub struct CcmAccounts {
+	pub cf_receiver: CcmAddress,
+	pub remaining_accounts: Vec<CcmAddress>,
+}
```

Comment:
I think we should not have the ```cf_receiver``` as part of the CcmAccounts type. The ```cf_receiver``` here is the same as the ```transfer_param.to```that is also passed in when creating the apicall so there is duplication. Moreover, we dont seem to use the ```cf_receiver``` when we construct the call. We use the ```transfer_param.to```. Also, its conceptuallz easier to follow if this type only contains the other accounts and so it can be renamed to the ```CcmTouchedAccounts``` or something.

Reply:
That's not right. The `cf_receiver` is not and should not be the same as the `transfer_params.to` because programs should not hold funds in Solana. It's therefore two different addresses.
And the `cf_receiver` is indeed used when crafting the `VaultProgram::ExecuteCcmNativeCall` instruction.
The naming `remaining_accounts` comes from the Anchor programs. Using a matching name was easy to understand (for me) but looking only at the State Chain it might not be the best naming. `CcmAdditionalAccounts` or something like that might be nicer.